ios/AppLovinMAXAdViewManager.m

```diff
@@ -227,18 +228,32 @@ - (void)attachAdViewIfNeededForViewTag:(NSNumber *)viewTag adUnitIdentifier:(NSS
         // If ad unit id and format has been set - create and attach AdView
         if ( [adUnitIdentifier al_isValidString] && adFormat )
         {
+            bool isNewAdView = NO;
```

Comment:
Can you walk me through the relevant call stack for when a mount and unmount happens? That way I can see if we really need all this code.

Reply:
Ok, let me try to explain what happens.

1. AppLovinMAXAdViewManager is created upon the app start.

2. When <AppLovinMAX.AdView .../> is added to the JS tree, AppLovinMAXAdViewManager creates a child view via createViewInstance() on Android, view() method on iOS.

3. When <AppLovinMAX.AdView .../> is removed from the JS tree, AppLovinMAXAdViewManager destroys the child view. On Android, onDropViewInstance() is called explicitly.  On iOS, the normal life cycle events such as didMoveToWindow() and removeFromSuperview() are called, and in the end, dealloc() is called.

Right now, our native AdView belongs to the child view and is created/destroyed at the same time as the child view.  In order to detect when the child view is destroyed in iOS,  I've added AppLovinMAXAdView.